packages/block-editor/src/components/inserter/search-items.js

```diff
@@ -20,7 +24,11 @@ const defaultGetCollection = () => null;
  * @return {Array} Words, extracted from the input string.
  */
 function extractWords( input = '' ) {
-	return noCase( input, {
+	if ( typeof extractedWords[ input ] !== 'undefined' ) {
+		return extractedWords[ input ];
```

Comment:
Let's use a `Map` instead, as the object will behave badly when `input` is `"valueOf"`, for example.

Also, in this type of code we usually try to minimize the number of map lookups (`cache[ input ]` calls). Do one read and one write, no more.

Reply:
Good point, done in f801b59cb286a2cb7711e79a22f17b1908bf0fba.